core/src/client/client.rs

```diff
@@ -449,6 +452,10 @@ impl BlockChainClient for Client {
             })
         })
     }
+
+    fn custom_handlers(&self) -> Vec<Arc<ActionHandler>> {
```

Comment:
Why noy `fn custom_handlers(&self) -> &[Arc<ActionHandler>>] {`

Reply:
If we return reference, then lock over `state_db` won't be released while holding that reference.
I thought it would be simpler to return a fresh copy of the vector and release the lock earlier instead of holding the lock.

Reply:
Okay. I see.